### PR TITLE
Tslint 4.0

### PR DESCRIPTION
**Pull request by jmlopez-rod**
This PR is not yet done. The CI will fail since `gulp-tslint` has not been released with the tslint update yet. It may happen sometime tomorrow.

For now all the changes included here make the tests pass if we skip the linting process:

```
gulp test --no-lint
```

I'll do another round of updates once this issue is resolved: https://github.com/panuhorsmalahti/gulp-tslint/issues/90

### CODE REVIEW

**Comment by jmlopez-rod** [comment]
@buzinas @blakeembrey All tests are passing again. Any of you have some time to do a quick code review? I'd like to release today so that I may start using tslint 4.0 and this project together.

File: tslint_eslint_rules.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (jmlopez-rod) [inline tslint_eslint_rules.json]: @buzinas I noticed that there is a file called `eslint_tslint.json` with a list of the rules in the project. Where those meant to be the preferences for the source code in this project? I tried them but it gave a lot of errors so I'll be adding some rules here and there as I see them fit. For now I started with the obvious ones. Is it safe to delete `eslint_tslint.json`?

**Comment by blakeembrey** [comment]
I think it looks good, thanks for your all your work!

**Comment by jmlopez-rod** [comment]
Awesome, i'll go ahead and release this sometime during my lunch break.

**Pull request merged**